File: solutions/typescript/resistor-color-duo/1/resistor-color-duo.ts
const bandColors: Record<string, number> = {
  "black": 0,
  "brown": 1,
  "red": 2,
  "orange": 3,
  "yellow": 4,
  "green": 5,
  "blue": 6,
  "violet": 7,
  "grey": 8,
  "white": 9
}

export function decodedValue(bands: string[]) {
  return 10 * bandColors[bands[0]]  + bandColors[bands[1]]
}

File: solutions/java/resistor-color-duo/1/src/main/java/ResistorColorDuo.java
import java.util.Arrays;
import java.util.List;

class ResistorColorDuo {
    private static final List<String> colors_ = Arrays.asList("black", "brown", "red", "orange", "yellow", "green",
            "blue", "violet", "grey", "white");

    int value(String[] colors) {
        return colors_.indexOf(colors[0]) * 10 + colors_.indexOf(colors[1]);
    }
}
